Library/Homebrew/descriptions.rb

```diff
@@ -123,9 +123,25 @@ def initialize(descriptions)
   # print them.
   def print
     blank = "#{Tty.yellow}[no description]#{Tty.reset}"
-    @descriptions.keys.sort.each do |name|
-      description = @descriptions[name] || blank
-      puts "#{Tty.white}#{name}:#{Tty.reset} #{description}"
+    @descriptions.keys.sort.each do |full_name|
+      short_name = formula_short_name(full_name)
+      printed_name = formula_name_counts[short_name] == 1 ? short_name : full_name
+      description = @descriptions[full_name] || blank
+      puts "#{Tty.white}#{printed_name}:#{Tty.reset} #{description}"
     end
   end
+
+  private
+
+  def formula_name_counts
+    @formula_name_counts ||= @descriptions.keys.each_with_object({}) do |full_name, counts|
+      name = formula_short_name(full_name)
+      counts[name] = (counts[name] || 0) + 1
+    end
+  end
+
+  def formula_short_name(full_name)
+    @formula_short_names ||= {}
+    @formula_short_names[full_name] ||= full_name.split('/')[-1]
```

Comment:
Maybe not, I figured it might be worth it to cut the number of `split` calls in half, especially if we're running `desc` with no arguments and iterating over all formulae in all taps.